tests/test_runner/test_discover_runner.py

```diff
@@ -297,6 +298,16 @@ def test_buffer_mode_test_fail(self):
         self.assertIn('Write to stderr.', stderr.getvalue())
         self.assertIn('Write to stdout.', stdout.getvalue())
 
+    @patch('faulthandler.enable')
+    def test_faulthandler_enabled(self, mocked_faulthandler):
+        _ = DiscoverRunner(no_faulthandler=False)
+        mocked_faulthandler.assert_called()
+
+    @patch('faulthandler.enable')
+    def test_faulthandler_disabled(self, mocked_faulthandler):
+        _ = DiscoverRunner(no_faulthandler=True)
+        mocked_faulthandler.assert_not_called()
```

Comment:
1. It's best practice to name mocks after the function
2. You don't need `_ =`. Python discards unused return values by default.

```suggestion
    @patch('faulthandler.enable')
    def test_faulthandler_enabled(self, mocked_enable):
        DiscoverRunner(no_faulthandler=False)
        mocked_enable.assert_called()

    @patch('faulthandler.enable')
    def test_faulthandler_disabled(self, mocked_enable):
        DiscoverRunner(no_faulthandler=True)
        mocked_enable.assert_not_called()
```